### ISSUE TITLE
Faculty course listing

### ISSUE THREAD

**Issue by kelslundberg** [betheluniversity/portal#89]
We need a channel in the portal to show active courses for faculty along with key links for each course.

I created a sample in the portal:
https://portal.bethel.edu/faculty-academics

Channel heading: 
Active Courses

Course number - Course name
Course location/Room number
Course links

Links for each course:
Moodle (course page in moodle)
Class List (banner)
Email Class (moodle)
Submit Grades (banner)

Here is also a screenshot of the sample content:
![screen shot 2017-06-26 at 4 00 58 pm](https://user-images.githubusercontent.com/6525391/27560127-ad84b592-5a88-11e7-83b5-e061dea6b127.png)

**Comment by e-jameson** [comment]
I ran this by Lisa this morning and and getting the courses shouldn't be a problem. I mentioned pulling in course URLs to Moodle and she said that might be more challenging, but probably do-able. Derek's done the most work on pulling Moodle data into Banner so we'll talk with him next week.

**Comment by e-jameson** [comment]
I met with Derek and he's going to work on this and get data it to us in the next week or so.

**Comment by kelslundberg** [comment]
@e-jameson Have we heard any updates on this from Derek?

**Comment by e-jameson** [comment]
Yup he's working on it right now. We should have data to put in relatively soon.

**Comment by dstavem** [comment]
See bth_portal_channel_api.class_list_fac.

This should get you most of what you need (and some of what you don't need).  Like the student version, I've included some extra columns for now until you're sure you have what you need, and then I can remove some of the extra ones.  For the Moodle link and the email class link, I think you can use the value in "mdl_crse_id" to create a URL (for example, something like https://moodle.bethel.edu/blocks/quickmail/email.php?courseid=20301).  Let me know if you need something different.  I still need to get the data for the class list and submit grades, but I think you should have everything else you'd need for this one.

**Comment by dstavem** [comment]
Note that for "location" tag (building and room), there are many courses that will be blank (e.g. internship, etc.), and there are others that will say "WEB" because they are web-only courses.  For those that are both web and in a building alternating (e.g. a hybrid course format), I took the first room that I found, even if it might be the second or third week that it meets there if the first meetings are web.  Let me know if you have questions as you dig into it.

**Comment by c-schwarze** [comment]
Thanks Derek! This is live on portal in the Faculty Academics tab

**Comment by kelslundberg** [comment]
We're waiting on Class Lists and Submit Grades....moving to In Process

**Comment by c-schwarze** [comment]
@e-jameson How are we intending to get Class Lists and Submit Grades?

**Comment by e-jameson** [comment]
We had an email thread about class lists last week, and @dstavem was going to look into an Argos report that we could call to generate a PDF with the class list, by passing in the ID Number of the course, or something along those lines.

Submit Grades is in Banner, need to figure out how the URL and how to generate it based on course data..

**Comment by dstavem** [comment]
I've added a field to the procedure bth_portal_channel_api.class_list_fac called "crse_roster_url."

This has a URL to an Argos report that should pull up a pdf roster of the students in the class with their photos.

I've done a bit of testing on my end, but please test it out a bit on your side, and let me know if you experience any issues.
